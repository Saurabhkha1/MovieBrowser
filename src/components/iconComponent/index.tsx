import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {IconProp} from '@fortawesome/fontawesome-svg-core';

type Props = {
  icon: IconProp;
  size?: number;
  color?: string;
};

export const IconComponent = ({icon, size, color}: Props) => {
  return <FontAwesomeIcon icon={icon} size={size} color={color} />;
};
