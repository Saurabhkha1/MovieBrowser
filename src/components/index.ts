export * from './iconComponent';
export * from './header';
export * from './movieComponent';
export * from './loader';
