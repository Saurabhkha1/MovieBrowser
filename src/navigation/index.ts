export * from './bottomTabs';
export * from './stack';
