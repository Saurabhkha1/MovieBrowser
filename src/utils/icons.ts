import {faSoundcloud} from '@fortawesome/free-brands-svg-icons';
import {
  faAngleLeft,
  faClose,
  faPeopleGroup,
  faPlay,
  faSearch,
  faStar,
} from '@fortawesome/free-solid-svg-icons';

export const Icons = {
  nowPlaying: faPlay,
  topRated: faStar,
  popular: faPeopleGroup,
  upcoming: faSoundcloud,
  backIcon: faAngleLeft,
  search: faSearch,
  close: faClose,
};
