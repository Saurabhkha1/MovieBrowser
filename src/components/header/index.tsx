import React, {useState, useRef, useEffect} from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Animated,
  TextInput,
  Keyboard,
} from 'react-native';
import {moderateScale} from '../../utils/helper';
import {Colors} from '../../utils/colors';
import {IconComponent} from '../iconComponent';
import {IconProp} from '@fortawesome/fontawesome-svg-core';
import {Icons} from '../../utils/icons';

type Props = {
  title: string;
  icon: IconProp;
  onChangeText: (text: string) => void;
  searchValue: string;
};

export const Header = ({title, icon, onChangeText, searchValue}: Props) => {
  const [isInputVisible, setInputVisible] = useState(false);
  const slideAnim = useRef(new Animated.Value(-100)).current;
  const inputRef = useRef<TextInput>(null);

  useEffect(() => {
    Animated.timing(slideAnim, {
      toValue: isInputVisible ? 0 : -100,
      duration: 300,
      useNativeDriver: true,
    }).start(() =>
      isInputVisible ? inputRef?.current?.focus() : Keyboard.dismiss(),
    );
  }, [isInputVisible, slideAnim]);

  const toggleInputVisibility = () => {
    setInputVisible(!isInputVisible);
    onChangeText?.('');
  };

  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        <IconComponent
          icon={icon}
          size={moderateScale(25)}
          color={Colors.white}
        />
        <Text style={styles.heading}>{title}</Text>
      </View>
      <TouchableOpacity onPress={toggleInputVisibility}>
        <IconComponent
          icon={Icons.search}
          size={moderateScale(25)}
          color={Colors.white}
        />
      </TouchableOpacity>

      <Animated.View
        style={[styles.inputContainer, {transform: [{translateY: slideAnim}]}]}>
        <TextInput
          ref={inputRef}
          style={styles.input}
          value={searchValue}
          placeholder="Search for movies..."
          placeholderTextColor={Colors.grey}
          onChangeText={onChangeText}
        />
        <TouchableOpacity
          style={styles.closeInputButton}
          onPress={toggleInputVisibility}>
          <IconComponent icon={Icons.close} size={moderateScale(25)} />
        </TouchableOpacity>
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: moderateScale(10),
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: Colors.darkBlue,
    justifyContent: 'space-between',
    position: 'relative',
  },
  leftContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  heading: {
    color: Colors.white,
    fontSize: moderateScale(20),
    fontWeight: '500',
    marginLeft: moderateScale(10),
  },
  inputContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: Colors.white,
    padding: moderateScale(5),
    paddingHorizontal: moderateScale(10),
    borderTopWidth: moderateScale(1),
    borderTopColor: Colors.grey,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  input: {
    height: moderateScale(40),
    borderColor: Colors.grey,
    borderWidth: moderateScale(1),
    borderRadius: moderateScale(5),
    paddingHorizontal: moderateScale(10),
    color: Colors.black,
    width: '90%',
  },
  closeInputButton: {
    width: '10%',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
