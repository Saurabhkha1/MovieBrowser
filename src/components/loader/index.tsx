import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React from 'react';
import {Colors} from '../../utils/colors';

type Props = {
  size?: 'full' | 'normal';
  color?: string;
};

export const Loader = ({size, color = Colors.black}: Props) => {
  return (
    <View style={size === 'full' ? styles.fullScreen : styles.normal}>
      <ActivityIndicator size={'large'} color={color} />
    </View>
  );
};

const styles = StyleSheet.create({
  fullScreen: {
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  normal: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
