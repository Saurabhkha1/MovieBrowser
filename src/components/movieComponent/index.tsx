import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  Platform,
} from 'react-native';
import React from 'react';
import {deviceWidth, moderateScale} from '../../utils/helper';
import {Colors} from '../../utils/colors';
import {Movie} from '../../redux/movieSlice';
import {IMAGE_BASE_URL} from '../../utils/constants';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../navigation';

type Props = {
  movie: Movie;
};

export const MovieComponent = ({movie}: Props) => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => navigation.navigate('MovieDetails', {movie: movie})}>
      <Image
        source={{
          uri: IMAGE_BASE_URL + movie?.poster_path,
        }}
        style={styles.poster}
      />
      <View style={styles.details}>
        <Text style={styles.name} numberOfLines={2} ellipsizeMode="tail">
          {movie?.title}
        </Text>
        <Text style={styles.dateWithNoMargin}>Release Date:</Text>
        <Text style={styles.date}>{movie?.release_date}</Text>
        <Text style={styles.date}>
          Rating: {movie?.vote_average?.toFixed(1)}/10
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    width: (deviceWidth - moderateScale(30)) / 2,
    height: Platform.OS === 'android' ? moderateScale(450) : moderateScale(400),
    borderRadius: moderateScale(10),
    backgroundColor: Colors.lightBlue,
    marginRight: moderateScale(10),
    marginBottom: moderateScale(10),
  },
  poster: {
    width: '100%',
    height: '70%',
    borderRadius: moderateScale(10),
  },
  details: {
    padding: moderateScale(5),
  },
  name: {
    color: Colors.black,
    fontSize: moderateScale(15),
    fontWeight: '500',
    marginBottom: moderateScale(10),
  },
  date: {
    color: Colors.black,
    fontSize: moderateScale(15),
    marginBottom: moderateScale(10),
  },
  dateWithNoMargin: {
    color: Colors.black,
    fontSize: moderateScale(15),
  },
});
