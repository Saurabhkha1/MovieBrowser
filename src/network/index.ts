import axios from 'axios';
import {BASE_URL, BEARER_TOKEN} from '../utils/constants';

type Movie = {
  id: number;
  title: string;
  poster_path: string;
  release_date: string;
  vote_average: number;
  overview: string;
};

export const fetchMovies = async (
  endpoint: string,
  page: number = 1,
): Promise<Movie[]> => {
  const response = await axios.get(`${BASE_URL}/movie/${endpoint}`, {
    params: {language: 'en-US', page},
    headers: {
      Authorization: `Bearer ${BEARER_TOKEN}`,
      Accept: 'application/json',
    },
  });
  return response.data.results;
};
