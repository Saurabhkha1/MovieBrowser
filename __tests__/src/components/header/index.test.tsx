import React from 'react';
import {create, act, ReactTestRenderer} from 'react-test-renderer';
import {Header} from '../../../../src/components';
import {Icons} from '../../../../src/utils/icons';
import {TextInput} from 'react-native';

const mockIcon = Icons.search;

jest.mock('../../../../src/screens', () => ({
  CommonHome: jest.fn(() => null),
}));

describe('Header Component', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    act(() => {
      jest.runOnlyPendingTimers();
      jest.useRealTimers();
    });
  });

  it('renders correctly', () => {
    let component: ReactTestRenderer | undefined;
    act(() => {
      component = create(
        <Header
          title="Test Title"
          icon={mockIcon}
          onChangeText={() => {}}
          searchValue=""
        />,
      );
    });
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows and hides input box on search icon press', () => {
    const onChangeTextMock = jest.fn();
    let component: ReactTestRenderer | undefined;
    act(() => {
      component = create(
        <Header
          title="Test Title"
          icon={mockIcon}
          onChangeText={onChangeTextMock}
          searchValue=""
        />,
      );
    });

    let tree = component?.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component?.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('updates input value correctly', () => {
    const onChangeTextMock = jest.fn();
    let component: ReactTestRenderer | undefined;
    act(() => {
      component = create(
        <Header
          title="Test Title"
          icon={mockIcon}
          onChangeText={onChangeTextMock}
          searchValue="Existing text"
        />,
      );
    });

    const input = component?.root?.findByType(TextInput);
    expect(input?.props.value).toBe('Existing text');
  });
});
