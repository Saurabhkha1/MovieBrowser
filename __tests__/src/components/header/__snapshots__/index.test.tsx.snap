// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "marginLeft": 10,
        }
      }
    >
      Test Title
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "#ccc",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "padding": 5,
        "paddingHorizontal": 10,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Search for movies..."
      placeholderTextColor="#ccc"
      style={
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "paddingHorizontal": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "width": "10%",
        }
      }
    >
      <MockedFontAwesomeIcon
        icon={
          {
            "icon": [
              384,
              512,
              [
                128473,
                10005,
                10006,
                10060,
                215,
                "close",
                "multiply",
                "remove",
                "times",
              ],
              "f00d",
              "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
            ],
            "iconName": "xmark",
            "prefix": "fas",
          }
        }
        size={25}
      />
    </View>
  </View>
</View>
`;

exports[`Header Component shows and hides input box on search icon press 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "marginLeft": 10,
        }
      }
    >
      Test Title
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "#ccc",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "padding": 5,
        "paddingHorizontal": 10,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Search for movies..."
      placeholderTextColor="#ccc"
      style={
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "paddingHorizontal": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "width": "10%",
        }
      }
    >
      <MockedFontAwesomeIcon
        icon={
          {
            "icon": [
              384,
              512,
              [
                128473,
                10005,
                10006,
                10060,
                215,
                "close",
                "multiply",
                "remove",
                "times",
              ],
              "f00d",
              "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
            ],
            "iconName": "xmark",
            "prefix": "fas",
          }
        }
        size={25}
      />
    </View>
  </View>
</View>
`;

exports[`Header Component shows and hides input box on search icon press 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "marginLeft": 10,
        }
      }
    >
      Test Title
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "#ccc",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "padding": 5,
        "paddingHorizontal": 10,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Search for movies..."
      placeholderTextColor="#ccc"
      style={
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "paddingHorizontal": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "width": "10%",
        }
      }
    >
      <MockedFontAwesomeIcon
        icon={
          {
            "icon": [
              384,
              512,
              [
                128473,
                10005,
                10006,
                10060,
                215,
                "close",
                "multiply",
                "remove",
                "times",
              ],
              "f00d",
              "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
            ],
            "iconName": "xmark",
            "prefix": "fas",
          }
        }
        size={25}
      />
    </View>
  </View>
</View>
`;

exports[`Header Component shows and hides input box on search icon press 3`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "500",
          "marginLeft": 10,
        }
      }
    >
      Test Title
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <MockedFontAwesomeIcon
      color="#fff"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={25}
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "#ccc",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "padding": 5,
        "paddingHorizontal": 10,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Search for movies..."
      placeholderTextColor="#ccc"
      style={
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000",
          "height": 40,
          "paddingHorizontal": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "width": "10%",
        }
      }
    >
      <MockedFontAwesomeIcon
        icon={
          {
            "icon": [
              384,
              512,
              [
                128473,
                10005,
                10006,
                10060,
                215,
                "close",
                "multiply",
                "remove",
                "times",
              ],
              "f00d",
              "M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z",
            ],
            "iconName": "xmark",
            "prefix": "fas",
          }
        }
        size={25}
      />
    </View>
  </View>
</View>
`;
